service/src/main/java/uk/nhs/adaptors/gp2gp/common/storageconnectors/AzureStorageConnector.java

```diff
@@ -0,0 +1,39 @@
+package uk.nhs.adaptors.gp2gp.common.storageconnectors;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import org.springframework.beans.factory.annotation.Autowired;
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.BlobContainerClient;
+
+public class AzureStorageConnector implements StorageConnector {
+
+    @Autowired
+    private BlobContainerClient containerClient;
+
+    protected AzureStorageConnector() {}
+
+    @Override
+    public void uploadToStorage(InputStream is, String filename) throws StorageConnectorException {
+        try {
+            BlobClient blobClient = containerClient.getBlobClient(filename);
+            blobClient.upload(is, is.available());
```

Comment:
Have a look at the javadoc for InputStream.available()

> Returns an estimate of the number of bytes that can be read (or skipped over) from this input stream without blocking

The impact of using this method would mean we might not upload the entire data stream!

I think our interface is going to need an additional 'length' parameter. In the future when we're streaming http response bodies directly into storage we can use the Content-Length header to set this parameter